pkg/cmd/util/clientcmd/factory.go

```diff
@@ -106,6 +107,15 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 		return describe.NewHumanReadablePrinter(noHeaders), nil
 	}
 
+	generators := map[string]kubectl.Generator{
+		"route/v1": routegen.RouteGenerator{},
```

Comment:
I think routes.  I think we'd want to have a copy or create-external-service function for service from service.

----- Original Message -----
> > @@ -106,6 +107,15 @@ func NewFactory(clientConfig kclientcmd.ClientConfig)
> > *Factory {
> >  		return describe.NewHumanReadablePrinter(noHeaders), nil
> >  	}
> >  
> > +	generators := map[string]kubectl.Generator{
> > +		"route/v1": routegen.RouteGenerator{},
> 
> @smarterclayton do you want `osc expose` to support exposing services from
> services or just routes?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2045/files#r30037558